changefeedccl: deflake tests that expect frequent span-level checkpoints

andyyang890 opened the pull request:
**changefeedccl: delete leftover comment**

This patch deletes a leftover comment that should've been deleted in
the commit that had aggregators periodically flush their progress.

Release note: None

---

**changefeedccl: deflake tests that expect frequent span-level checkpoints**

In a recent commit, the interval at which change aggregators would
flush their frontiers to the change frontier was decoupled from
the span-level checkpoint interval (which is configurable via a
cluster setting). This caused some tests that only set the cluster
setting (and not `min_checkpoint_frequency`) to a low value to
sometimes flake. This patch updates all such tests to also configure
the `min_checkpoint_frequency` option.

[master]
Fixes #151279

[release-25.4]
Informs #154066

Release note: None

---

**changefeedccl: deflake initial backfill checkpoint tests**

This patch deflakes the initial backfill checkpoint tests, which filter
out resolved spans to force gaps during a backfill. The tests however
did not ensure that there would be at least one span that was not
filtered out, which would result in no span-level checkpoint being
created, leading the tests to time out.

Release note: None

andyyang890 committed:
changefeedccl: delete leftover comment

This patch deletes a leftover comment that should've been deleted in
the commit that had aggregators periodically flush their progress.

Release note: None

andyyang890 committed:
changefeedccl: deflake tests that expect frequent span-level checkpoints

In a recent commit, the interval at which change aggregators would
flush their frontiers to the change frontier was decoupled from
the span-level checkpoint interval (which is configurable via a
cluster setting). This caused some tests that only set the cluster
setting (and not `min_checkpoint_frequency`) to a low value to
sometimes flake. This patch updates all such tests to also configure
the `min_checkpoint_frequency` option.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154143)
<!-- Reviewable:end -->

andyyang890 committed:
changefeedccl: deflake initial backfill checkpoint tests

This patch deflakes the initial backfill checkpoint tests, which filter
out resolved spans to force gaps during a backfill. The tests however
did not ensure that there would be at least one span that was not
filtered out, which would result in no span-level checkpoint being
created, leading the tests to time out.

Release note: None

aerfrei commented on pkg/ccl/changefeedccl/alter_changefeed_test.go:
I'm seeing it for the second commit here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/20518187?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&expandBuildTestsSection=true&hideProblemsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true.

andyyang890 replied:
Yeah that's the issue that the third commit is fixing, which I'm referencing in https://github.com/cockroachdb/cockroach/pull/154143#discussion_r2379773635

andyyang890 commented on pkg/ccl/changefeedccl/alter_changefeed_test.go:
The one in the PR description (#151279) and previously #153918

asg0451 commented on pkg/ccl/changefeedccl/changefeed_test.go:
```diff
@@ -9115,22 +9117,21 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
+		var allowedOne, haveGaps bool
 		knobs.FilterSpanWithMutation = func(r *jobspb.ResolvedSpan) (bool, error) {
 			if r.Span.Equal(tableSpan) {
-				// Do not emit resolved events for the entire table span.
```
why remove this bit? seems potentially useful

andyyang890 replied:
I had to remove the analogous piece of code in `TestAlterChangefeedAddTargetsDuringBackfill` in #153758 because it was causing rare flakes where this fake span got flushed to the change frontier and showed up in the checkpoint, which caused a test assertion to fail. Preemptively removing it here in case it causes similar flakes.
